base/applications/rapps/appdb.cpp

```diff
@@ -141,74 +141,124 @@ CAppDB::UpdateAvailable()
     EnumerateFiles();
 }
 
-VOID
-CAppDB::UpdateInstalled()
+static inline HKEY
+GetRootKeyInfo(UINT Index, REGSAM &RegSam)
 {
-    // Remove all old entries
-    ClearList(m_Installed);
+    C_ASSERT(_countof(g_RootKeyEnum) == _countof(g_RegSamEnum));
+    if (Index < _countof(g_RootKeyEnum))
+    {
+        RegSam = g_RegSamEnum[Index];
+        return g_RootKeyEnum[Index];
+    }
+    return NULL;
+}
 
-    int LoopKeys = 2;
+HKEY
+CAppDB::EnumInstalledRootKey(UINT Index, REGSAM &RegSam)
+{
+    // Loop for through all combinations.
+    // Note that HKEY_CURRENT_USER\Software does not have a redirect
+    // https://docs.microsoft.com/en-us/windows/win32/winprog64/shared-registry-keys#redirected-shared-and-reflected-keys-under-wow64
+    if (Index < (IsSystem64Bit() ? 3 : 2))
+        return GetRootKeyInfo(Index, RegSam);
+    else
+        return NULL;
+}
 
-    if (IsSystem64Bit())
+CInstalledApplicationInfo *
+CAppDB::CreateInstalledAppByRegistryKey(LPCWSTR KeyName, HKEY hKeyParent, UINT KeyIndex)
+{
+    CRegKey hSubKey;
+    if (hSubKey.Open(hKeyParent, KeyName, KEY_READ) == ERROR_SUCCESS)
```

Comment:
```suggestion
    if (hSubKey.Open(hKeyParent, KeyName, KEY_READ) != ERROR_SUCCESS)
        return NULL;
```
else continue as before, but with one indentation level less.